pymc_marketing/mmm/base.py

```diff
@@ -393,24 +415,206 @@ def plot_contribution_curves(self) -> plt.Figure:
+        - The channel contributions are computed using the `compute_channel_contribution_original_scale` method.
+        - The lower and upper bounds for the budget allocation are determined based on the provided `budget_bounds` or default values.
+        - Optimization is performed to find the optimal budget allocation that maximizes contribution while satisfying the budget constraint.
+        - The results are stored in a dictionary containing the allocated budget and contribution information.
+        - The updated DataFrame with the allocated budget and contribution information is returned as part of the dictionary.
+        """
+
+        if total_budget is None or df_estimations is None:
+            raise ValueError(
+                "Total Budget and Estimations dataframe parameters must be provided."
+            )
+
+        if total_budget is None:
+            total_budget = {}
+
+        if not isinstance(total_budget, (int, float)):
+            raise ValueError(
+                "The 'total_budget' parameter must be an integer or float."
+            )
+
+        if not isinstance(df_estimations, pd.DataFrame):
+            raise ValueError(
+                "The 'df_estimations' parameter must be a pandas DataFrame."
+            )
+
+        if not isinstance(budget_bounds, dict):
+            raise ValueError("The 'budget_bounds' parameter must be a dictionary.")
+
+        df = df_estimations.copy()
+        channel_contributions = self.compute_channel_contribution_original_scale().mean(
+            ["chain", "draw"]
+        )
+
+        lower_bounds = []
+        upper_bounds = []
+
+        for channel in df["channel"]:
+            if channel in budget_bounds:
+                min_bound, max_bound = budget_bounds[channel]
+            else:
+                min_bound = np.quantile(self.data[channel], 0.25)
+                max_bound = df.loc[df["channel"] == channel, "plateau_point"].values[0]
+
+            lower_bounds.append(min_bound)
+            upper_bounds.append(max_bound)
+
+        bounds = Bounds(lower_bounds, upper_bounds)
+        cons = {"type": "eq", "fun": budget_constraint, "args": (total_budget,)}
+
+        initial_guess = lower_bounds  # Start with minimum allocation for all channels
+        result = minimize(
+            cost_function,
+            initial_guess,
+            args=(df, self.data, channel_contributions),
+            bounds=bounds,
+            constraints=cons,
+        )
+
+        # Add the new column to the dataframe
+        df["Optimal Budget"] = np.round(result.x)
+
+        contributions = pd.DataFrame(
+            {
+                "estimated_total_budget": np.round(result.x.sum()),
+                "estimated_max_contribution": (-result.fun),
+            },
+            index=[0],
+        )
+
+        result_dict = {"allocation": df, "contribution": contributions}
```

Comment:
you can simply `return {"allocation": df, "contribution": contributions}` so that we do not have unnecessary temporal variables ;)